tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNetCore/AspNetCoreBlockMiddlewareIntegration.cs

```diff
@@ -0,0 +1,109 @@
+﻿// <copyright file="AspNetCoreBlockMiddlewareIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if !NETFRAMEWORK
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Extensions.DependencyInjection;
+#endif
+using System;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Datadog.Trace.AppSec;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AspNetCore
+{
+    /// <summary>
+    /// The ASP.NET Core middleware integration.
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "Microsoft.AspNetCore.Hosting",
+        TypeName = "Microsoft.AspNetCore.Hosting.Builder.ApplicationBuilderFactory",
+        MethodName = "CreateBuilder",
+        ParameterTypeNames = new[] { "Microsoft.AspNetCore.Http.Features.IFeatureCollection" },
+        ReturnTypeName = "Microsoft.AspNetCore.Builder.IApplicationBuilder",
+        MinimumVersion = "2",
+        MaximumVersion = "6",
+        IntegrationName = nameof(IntegrationId.AspNetCore))]
+    public class AspNetCoreBlockMiddlewareIntegration
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(AspNetCoreBlockMiddlewareIntegration));
+
+        /// <summary>
+        /// test
+        /// </summary>
+        /// <param name="instance">instance</param>
+        /// <param name="returnValue">returnValue</param>
+        /// <param name="exception">exception</param>
+        /// <param name="state">state</param>
+        /// <typeparam name="TTarget">TTarget</typeparam>
+        /// <typeparam name="TReturn">TReturn</typeparam>
+        /// <returns>CallTargetReturn</returns>
+        public static CallTargetReturn<TReturn> OnMethodEnd<TTarget, TReturn>(TTarget instance, TReturn returnValue, Exception exception, in CallTargetState state)
+        {
+#if !NETFRAMEWORK
+            if (Security.Instance.Settings.Enabled)
+            {
+                var appb = (IApplicationBuilder)returnValue;
```

Comment:
I don't know if it should, but the direct reference to the microsoft libs that we have make me a little nervous 😬 I can't think of a scenario where it will be an issue, seeing as we _already_ reference this libs, but I wonder if we should use duck typing here to be on the safe side? 🤔 

EDIT: It's fine, we reference loads of stuff in the diagnostic observer, which seems risky, but hasn't bitten us yet AFAIK!